packages/peregrine/lib/hooks/useCarousel.js

```diff
@@ -1,62 +1,40 @@
-import { useCallback, useReducer, useMemo } from 'react';
-import withLogger from '../util/withLogger';
+import { useCallback, useMemo, useState } from 'react';
 
 const sortImages = (images = []) =>
     images
         .filter(({ disabled }) => !disabled)
         .sort((a, b) => a.position - b.position);
 
-const reducer = (state, action) => {
-    switch (action.type) {
-        case 'previous':
-            if (state.index > 0) {
-                return { ...state, index: state.index - 1 };
-            } else {
-                return { ...state, index: state.length - 1 };
-            }
-        case 'next':
-            return { ...state, index: (state.index + 1) % state.length };
-        case 'set':
-            return { ...state, index: action.value };
-        default:
-            throw new Error('Unknown carousel action.');
-    }
-};
-
-const wrappedReducer = withLogger(reducer);
-
 /**
  * A hook for interacting with the state of a carousel of images.
  *
  * @param {Array} images an array of image objects
  * @param {number} startIndex the index at which to start the carousel
  */
 export const useCarousel = (images = [], startIndex = 0) => {
-    const [{ index }, dispatch] = useReducer(wrappedReducer, {
-        index: startIndex,
-        length: images.length
-    });
-    const sortedImages = useMemo(() => sortImages(images), [images]);
+    const [activeItemIndex, setActiveItemIndex] = useState(startIndex);
 
-    const handlePrevious = useCallback(
-        () => dispatch({ type: 'previous' }),
-        []
-    );
+    const sortedImages = useMemo(() => sortImages(images), [images]);
 
-    const handleNext = useCallback(() => dispatch({ type: 'next' }), []);
+    const handlePrevious = useCallback(() => {
+        if (activeItemIndex > 0) {
+            setActiveItemIndex(activeItemIndex - 1);
+        } else {
+            setActiveItemIndex(images.length - 1);
+        }
+    }, [activeItemIndex, images]);
```

Comment:
When using `useCallback`, look out for scenarios like this one. As written, `handlePrevious` will be recreated whenever `activeItemIndex` changes, and since `handlePrevious` itself changes `activeItemIndex`, this memoization is pointless.

Before, we were keeping `index` in a reducer so that `handlePrevious` could be memoized. If you're going to remove the reducer in favor of `useState`, you'll want to use the [functional update](https://reactjs.org/docs/hooks-reference.html#functional-updates) form.

```es6
const handlePrevious = useCallback(() => {
  setActiveItemIndex(prev => prev > 0 ? prev - 1 : images.length - 1)
}, [images])
```